dids/diddht/diddht.go

```
@@ -1,175 +1,191 @@
+		VerificationMethod: []didcore.VerificationMethod{},
+	}
+
+	// 4. Construct a conformant JSON representation of a DID Document.
+	for _, vm := range o.verificationMethods {
+		purposes, ok := o.purposes[vm.ID]
+		if !ok {
+			purposes = []didcore.Purpose{}
 		}
-		didRecord.records[h.Name.String()] = fullTxtRecord
+		document.AddVerificationMethod(vm, didcore.Purposes(purposes...))
 	}
 
-	return &didRecord, nil
-}
+	for _, s := range o.services {
+		document.AddService(s)
+	}
 
-// TODO on the diddhtrecord we should validate the minimum reqs for a valid did
-func parseVerificationRelationships(rootRecord string) (map[string][]string, error) {
-	rootRecordProps, err := parseTXTRecordData(rootRecord)
+	// 5. Map the output DID Document to a DNS packet
+	msgBytes, err := dns.MarshalDIDDocument(&document)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to marshal did document to dns packet: %w", err)
 	}
-	// reverse the map to get the relationships
-	relationshipMap := map[string][]string{}
-	for k, values := range rootRecordProps {
-		v := strings.Join(values, "")
-		rel, ok := relationshipMap[v]
-		if !ok {
-			rel = []string{}
-		}
-		rel = append(rel, k)
-		relationshipMap[v] = rel
+
+	// 6. Construct a signed BEP44 put message with the v value as a bencoded DNS packet from the prior step.
+	seq := time.Now().Unix() / 1000
+
+	signer := func(payload []byte) ([]byte, error) {
+		return keyMgr.Sign(keyID, payload)
 	}
 
-	return relationshipMap, nil
-}
+	bep44Msg, err := bep44.NewSignedBEP44Message(msgBytes, seq, publicKeyBytes, signer)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create signed bep44 message: %w", err)
+	}
 
-func parseTXTRecordData(data string) (map[string][]string, error) {
-	result := map[string][]string{}
-	fields := strings.Split(data, ";")
-	if len(fields) == 0 {
-		return nil, errors.New("no fields found")
+	if o.relay == nil {
+		return nil, errors.New("no relay provided")
```

we can pull this up to the top of the function so we can fail early if the right stuff isn't provided